### PR TITLE
Annotation-work-/File-Cleaning-part-8

### PR DESCRIPTION
**Pull request by MrRowey**
Files from the `/lua/*` Directory
- ScenarioPlatoonAI.lua
- kirvesweapons.lua
- lazyvar.lua
- objectiveArrow.lua
- platoon.lua
- platoontemplates.lua
- proptree.lua
- scenarioEnvironment.lua
- selfdestruct.lua
- seraphimprojectiles.lua

### CODE REVIEW

File: lua/platoon.lua [hunk 1]
```diff
@@ -6588,6 +6661,11 @@ Platoon = Class(moho.platoon_methods) {
     -- @param path A table of positions, preferably of type Vector. Converted otherwise.
     -- @param self.PlatoonData.UseFormation The formation to apply, such as GrowthFormation, AttackFormation or NoFormation.
     -- @return Table of commands.
+    ---comment
+    ---@param self any
+    ---@param path any
+    ---@param formation any
+    ---@return table
     IssuePatrolAlongRoute = function(self, path, formation)
```
Comment (Hdt80bro) [inline lua/platoon.lua:6669]: path is `Vector[]`
formation is an optional string
return is `PlatoonCommand[]`

File: lua/platoon.lua [hunk 2]
```diff
@@ -6669,10 +6747,10 @@ Platoon = Class(moho.platoon_methods) {
...
 
     --- Aggressive-moves the platoon along the path, orientating at each node to match the line from the previous node to the current node.
-    -- @param self The platoon itself.
-    -- @param path A table of positions, preferably of type Vector. Converted otherwise.
-    -- @param self.PlatoonData.UseFormation The formation to apply, such as GrowthFormation, AttackFormation or NoFormation.
-    -- @return Table of commands.
+    ---@param self Platoon
+    ---@param path Vector[]
+    ---@param formation string[]
+    ---@return table
```
Comment (Hdt80bro) [inline lua/platoon.lua:6753]: same as above

File: lua/platoon.lua [hunk 3]
```diff
@@ -6760,6 +6838,10 @@ Platoon = Class(moho.platoon_methods) {
     -- @param path A table of positions, preferably of type Vector. Converted otherwise.
     -- @param self.PlatoonData.UseFormation The formation to apply, such as GrowthFormation, AttackFormation or NoFormation.
     -- @return Table of commands.
+    ---@param self Platoon
+    ---@param path Vector[]
+    ---@param formation string[]
+    ---@return table
```
Comment (Hdt80bro) [inline lua/platoon.lua:6844]: same also

**Review by Hdt80bro — COMMENTED** [review]
There's quite a few annotation type we haven't created yet; don't worry about them, we can come back to those later

File: lua/platoon.lua (hunk 1 above)
Comment (Garanas) [reply lua/platoon.lua:6669]: Formation is `UnitFormations` , could be optional

File: lua/lazyvar.lua [hunk 4]
```diff
@@ -49,6 +49,7 @@ function LazyVarMetaTable:__call()
     return self[1]
```
[the remark this answers is not in the document]
Comment (Garanas) [reply lua/lazyvar.lua:49]: I wonder who that person is 😄

File: lua/proptree.lua [hunk 5]
```diff
@@ -50,11 +51,20 @@ Tree = Class(Prop) {
...
+    ---@param self Tree
+    ---@param other string
+    ---@return boolean
     OnCollisionCheck = function(self, other)
         return not self.Dead
     end,
 
     --- Collision check with units
+    ---@param self Tree
+    ---@param other string
```
[the remark this answers is not in the document]
Comment (Garanas) [reply lua/proptree.lua:63]: other is a unit here, not a projectile

File: lua/platoon.lua (hunk 1 above)
Comment (Hdt80bro) [reply lua/platoon.lua:6669]: It's optional, two times over :)
`local formation = formation or self.PlatoonData.UseFormation or 'NoFormation'`

File: lua/lazyvar.lua (hunk 4 above)
[the remark this answers is not in the document]
Comment (Hdt80bro) [reply lua/lazyvar.lua:49]: I was planning on it, but I actually believe 4z0t said he had something in mind.

**Pull request merged**